week6/Filmer/app/src/main/java/com/tahhan/filmer/viewmodel/MovieViewModel.kt

```diff
@@ -1,40 +1,32 @@
 package com.tahhan.filmer.viewmodel
 
 import android.app.Application
-import androidx.lifecycle.AndroidViewModel
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import com.google.firebase.auth.FirebaseAuth
+import androidx.lifecycle.*
 import com.tahhan.filmer.model.Movie
 import com.tahhan.filmer.repository.MovieRepo
-import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 
 class MovieViewModel(application: Application) : AndroidViewModel(application) {
     private val repo = MovieRepo(application)
+    fun getMovies(): LiveData<List<Movie>> = repo.getPopularMovieList()
 
-    fun getMovies(): LiveData<List<Movie>> {
-        return repo.getPopularMovieList()
-    }
 
-    fun getFavoriteMovieList(): LiveData<List<Movie>> {
-        return repo.getFavoriteMovieList()
-    }
+    fun getFavoriteMovieList(): LiveData<List<Movie>> =
+        repo.getFavoriteMovieList().asLiveData()
 
-    fun getFavoriteMovieByID(id: String):  MutableLiveData<Movie>? {
 
-        return repo.getFavoriteMovieByID(id)
-    }
+    fun getFavoriteMovieByID(id: String): LiveData<Movie> =
+        repo.getFavoriteMovieByID(id).asLiveData()
 
-    fun insertMovie(movie: Movie) {
-            repo.insertMovie(movie)
-    }
 
-    fun removeMovie(movie: Movie) {
-        repo.removeMovie(movie)
+    fun insertMovie(movie: Movie) = viewModelScope.launch(Dispatchers.IO) {
```

Comment:
You don't need to use dispatchers here, you can just call insert and the suspend modifier will take care of the rest